src/preamble.js

```diff
@@ -476,6 +476,7 @@ function addOnPostRun(cb) {
 var runDependencies = 0;
 var runDependencyWatcher = null;
 var dependenciesFulfilled = null; // overridden to take different actions when all run dependencies are fulfilled
+var waitDependenciesResolves = [];
```

Comment:
Can we use this to replace the existing `dependenciesFulfilled` thing above?  And use this internally?

Reply:
Sure.

Reply:
CI seems unhappy since Promise is not an ES5 thing. In practice it will run fine because Promises are only used when waitRunDependencies is explicitly called, and the caller would be expecting a Promise if they were to call such a function.